prometheus-openstack-exporter

```diff
@@ -253,7 +253,8 @@ class Neutron():
         owner = "network:router_gateway"
         for port in self.ports:
             if port["device_id"] == uuid and port["device_owner"] == owner:
-                return port["fixed_ips"][0]["ip_address"]
+                if port["status"] == "ACTIVE" and port["fixed_ips"] is not None:
```

Comment:
Can port["fixed_ips"] be an empty list? s/ is not None// would cover that handle that case too.

Reply:
yes, most definitely...will fix that to be an array length check.